classes/author/Collector.inc.php

```diff
@@ -0,0 +1,232 @@
+<?php
+/**
+ * @file classes/author/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class author
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
+ */
+
+namespace PKP\author;
+
+use Illuminate\Database\Query\Builder;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
+use PKP\plugins\HookRegistry;
+
+class Collector implements CollectorInterface
+{
+    public const ORDERBY_SEQUENCE = 'sequence';
+    public const ORDERBY_ID = 'id';
+
+    /** @var string The default orderBy value for authors collector */
+    public $orderBy = self::ORDERBY_SEQUENCE;
+
+    /** @var DAO */
+    public $dao;
+
+    /** @var array|null */
+    public $contextIds = null;
+
+    /** @var array|null */
+    public $publicationIds = null;
+
+    /** @var string get authors with a family name */
+    protected $familyName = null;
+
+    /** @var string get authors with a given name */
+    protected $givenName = null;
+
+    /** @var string get authors with a specified country code */
+    protected $country = null;
+
+    /** @var string get authors with a specified affiliation */
+    protected $affiliation = null;
+
+    /** @var int */
+    public ?int $count;
+
+    /** @var int */
+    public ?int $offset;
+
+    /** @var bool|null */
+    public $includeInBrowse = null;
+
+    public function __construct(DAO $dao)
+    {
+        $this->dao = $dao;
+    }
+
+    /**
+     * Filter by contexts
+     */
+    public function filterByContextIds(?array $contextIds): self
+    {
+        $this->contextIds = $contextIds;
+        return $this;
+    }
+
+    /**
+     * Filter by publications
+     */
+    public function filterByPublicationIds(?array $publicationIds): self
+    {
+        $this->publicationIds = $publicationIds;
+        return $this;
+    }
+
+    /**
+     * Filter by include in browse
+     */
+    public function filterByIncludeInBrowse(bool $includeInBrowse): self
+    {
+        $this->includeInBrowse = $includeInBrowse;
+        return $this;
+    }
+
+    /**
+     * Include orderBy columns to the collector query
+     */
+    public function orderBy(string $orderBy): self
+    {
+        $this->orderBy = $orderBy;
+        return $this;
+    }
+
+    /**
+     * Filter by the given and family name
+     *
+     *
+     */
+    public function filterByName(string $givenName, string $familyName): self
```

Comment:
These parameters (and others in setters like this) should be nullable (`?string`) in order to reset the constraints if desired.

Reply:
Fixed